### PR TITLE
Friendships v2 - Milestone 3

### PR DESCRIPTION
**Pull request by flpfar**
This PR adds the following:
- Accepts mutual friendships - change friendships creation and update.
- New method in user model
- Add link to accept invitation in all users page
- Tests for mutual friendships and view changes

### CODE REVIEW

**Comment by pelzolga123** [comment]
## Hi Team!
### Your project looks great :+1: 
Still, there are some changes required before this Pull Request can be approved:
* [x] Posts timeline should include posts written by logged-in user and their friends, as described [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png). Right now I can see posts from users, that are not current user friends.
* [x] Make sure to move tests for Friendship in a separate file. Also, your tests are missing the next test cases:
* Tests should make sure the user avoids self friendship.
* Tests should make sure the user can see only their posts and post of their friends on the Timeline.

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

**Comment by flpfar** [comment]
Thanks for reviewing our code, @pelzolga123 ! All changes requested were implemented, with respective tests. We really appreciate your help! :+1:

File: spec/features/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  before(:each) do
+    visit new_user_session_path
+    fill_in 'user_email', with: 'mauricio@mail.com'
+    fill_in 'user_password', with: '123456'
+    find("input[type='submit']").click
+  end
+
+  it 'when -mauricio@mail.com- sign in, he can his posts' do
+    expect(page).to have_content(@mauricios_post)
+  end
```
Comment (euqueme) [inline spec/features/post_spec.rb]: [OPTIONAL]
- [x] I think this it block can be changed to be more general like: "when a User signs in, he/she can see his/her posts"

File: spec/features/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  it 'when -mauricio@mail.com- sign in, he can see the post of -augusto@mail.com-' do
+    expect(page).to have_content(@augustos_post)
+  end
```
Comment (euqueme) [inline spec/features/post_spec.rb]: [OPTIONAL]

- [x] Also the description of this `it` block can be something like: 'when a User signs in, he/she can see posts from his/her friends'

File: spec/features/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  it 'when -mauricio@mail.com- sign in, he can not see the post of -magic@mail.com-' do
+    expect(page).to_not have_content(@magic_post)
+  end
```
Comment (euqueme) [inline spec/features/post_spec.rb]: [OPTIONAL]

- [x] Finally, the description of this test case could be changed to "when the user signs in, he/she cannot see posts from users that aren't his/her friends"

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @flpfar and @sergiomauz :handshake: 

### Thank you for working on the changes requested by the previous reviewer, :medal_sports: :medal_sports:  I like that you created a section for friendships, your project is working perfectly :ok_hand: so you may merge this branch and continue :tada: :tada: :tada: 

[OPTIONAL]

- [x] I strongly recommend to change the description of some of your test cases to make them more general and using the concepts of your project like users, friends, etc.. I've left some examples in your code below :point_down: 

### Keep up the good work :clinking_glasses:

Any question? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: APPROVED :green_apple:

**Comment by flpfar** [comment]
Thank you for reviewing our code, @euqueme ! Thanks for the recommendation of optional changes. We implemented them and that's ok now.

**Pull request merged**